api/all/src/main/java/io/opentelemetry/api/internal/InternalAttributeKeyImpl.java

```diff
@@ -52,13 +57,23 @@ public String getKey() {
     return key;
   }
 
+  /** Returns the key, encoded as UTF-8 bytes. */
+  public byte[] getKeyUtf8() {
+    byte[] keyUtf8 = this.keyUtf8;
```

Comment:
Are we ok with not having this field be volatile, and just having multiple threads recalculate if necessary? I think that should be fine, since it'll still be better than computing every time on every thread, and avoiding extra memory barriers is also good.

Reply:
Yup, and in practice with a BSP that doesn't happen anyways. It's same as String.hashCode